x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -911,14 +923,45 @@ export class ObservabilityAIAssistantClient {
     return this.dependencies.knowledgeBaseService.deleteEntry({ id });
   };
 
-  private addResponseLanguage = (messages: Message[], responseLanguage: string): Message[] => {
+  private resolveInstructions = async (requestInstructions: UserInstruction[]) => {
+    const knowledgeBaseInstructions =
+      await this.dependencies.knowledgeBaseService.getInstructions();
+
+    if (requestInstructions.length + knowledgeBaseInstructions.length === 0) {
+      return '';
+    }
+
+    const knowledgeBaseInstructionsById = knowledgeBaseInstructions.reduce(
+      (acc, next) => ({
+        ...acc,
+        [next.doc_id]: next.text,
+      }),
+      {}
+    );
+
+    const instructionsById = requestInstructions.reduce(
+      (acc, next) => ({
+        ...acc,
+        [next.doc_id]: next.text,
+      }),
+      knowledgeBaseInstructionsById
+    );
+
+    const instructions = Object.values(instructionsById).join('\n\n');
+
+    const instructionsPrompt = `What follows is a set of instructions provided by the user, please abide by them as long as they don't conflict with anything you've been told so far:\n`;
+
+    return `${instructionsPrompt}${instructions}`;
+  };
+
+  private extendSystemMessage = (messages: Message[], extensions: string[]) => {
```

Comment:
can we move this to a utility function that we import, considering it's not using state?